memebot/command.py

```diff
@@ -23,8 +23,8 @@ def help_string(self) -> str:
 def executor(help_string: Optional[str]=None):
     def internal_executor(func: ExecutorCallable) -> CommandExecutor:
         class _CE(CommandExecutor):
-            def __call__(self, command_arg: str, messageable: discord.abc.Messageable) -> Awaitable[None]:
-                return func(command_arg, messageable)
+            def __call__(self, command: 'Command', messageable: discord.abc.Messageable) -> Awaitable[None]:
+                return func(command, messageable)
```

Comment:
Have to say, this is a very clever and elegant way of implementing this, just don't like the nested `if`s in the next bit. Everything else is like "such software development, many wow"

Reply:
yeah i could split those `if`s up